### PR TITLE
Reformat unset_default_dtype

### LINKED ISSUE #4277: unset_default_dtype

### PR DESCRIPTION
**Pull request by AbdulMuheeth**
Added short description and examples to the docstrings.
 resolves #4277

### CODE REVIEW

File: ivy/functional/ivy/data_type.py [hunk 1]
```diff
@@ -1963,7 +1963,35 @@ def type_promote_arrays(
...
+    """
+    Unsets the datatype dtype from default data type.
+
+    Parameters
+    ----------
+    None
+
+    Examples
+    --------
+
```
Comment (jkeane508) [inline ivy/functional/ivy/data_type.py:1975]: There shouldn't be a blank line here according to numpy style docstring test

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Small fix required

File: ivy/functional/ivy/data_type.py (hunk 1 above)
Comment (AbdulMuheeth) [reply ivy/functional/ivy/data_type.py:1975]: Ok, I will change it.

Comment (AbdulMuheeth) [reply ivy/functional/ivy/data_type.py]: @jkeane508 made suggested changes. Please let me know if any other changes need to be done.

Comment (jkeane508) [reply ivy/functional/ivy/data_type.py]: Yea this looks good to me!

**Review by jkeane508 — APPROVED** [review]
LGTM!

**Pull request merged**